kernel/src/main/java/org/kframework/parser/inner/kernel/KSyntax2Bison.java

```diff
@@ -387,60 +388,79 @@ private static void processProduction(Production prod, Module module, Module dis
           "  node *n = malloc(sizeof(node) + sizeof(node *));\n" +
           "  n->str = false;\n" +
           "  n->location = @$;\n" +
-          "  n->hasLocation = " + (hasLocation ? "1" : "0") + ";\n" +
-          "  n->nchildren = 1;\n" +
-          "  n->sort = \"");
-      encodeKore(prod.sort(), bison);
-      bison.append("\";\n" +
-          "  if (!$1.nterm->str && strncmp($1.nterm->symbol, \"inj{\", 4) == 0) {\n" +
-          "    char *childSort = $1.nterm->children[0]->sort;\n" +
-          "    n->symbol = injSymbol(childSort, n->sort);\n" +
-          "    n->children[0] = $1.nterm->children[0];\n" +
-          "  } else {\n" +
-          "    n->symbol = \"inj{");
-      encodeKore(prod.getSubsortSort(), bison);
-      bison.append(", ");
-      encodeKore(prod.sort(), bison);
-      bison.append("}\";\n" +
-          "    n->children[0] = $1.nterm;\n" +
-          "  }\n");
-      if (prod.att().contains("userListTerminator")) {
-        KLabel nil = KLabel(prod.att().get("userListTerminator"));
-        KLabel cons = KLabel(prod.att().get("userList"));
-        bison.append(
-          "  node *n2 = malloc(sizeof(node));\n" +
-          "  n2->symbol = \"");
-        encodeKore(nil, bison);
-        bison.append("\";\n" +
-          "  n2->str = false;\n" +
-          "  n2->location = @$;\n" +
-          "  n2->hasLocation = 0;\n" +
-          "  n2->nchildren = 0;\n" +
-          "  n2->sort = \"");
+          "  n->hasLocation = " + (hasLocation ? "1" : "0") + ";\n");
+      if (prod.sort().equals(Sorts.K()) && prod.getSubsortSort().equals(Sorts.KItem())) {
+        bison.append("  n->nchildren = 2;\n" +
+            "  n->sort = \"");
         encodeKore(prod.sort(), bison);
         bison.append("\";\n" +
-          "  node *n3 = malloc(sizeof(node) + 2*sizeof(node *));\n" +
-          "  n3->symbol = \"");
-        encodeKore(cons, bison);
+            "  n->symbol = \"kseq{}\";\n" +
+            "  n->children[0] = $1.nterm;\n" +
+            "  node *n2 = malloc(sizeof(node));\n" +
+            "  n2->symbol = \"dotk{}\";\n" +
+            "  n2->str = false;\n" +
+            "  n2->location = @$;\n" +
+            "  n2->hasLocation = false;\n" +
+            "  n2->nchildren = 0;\n" +
+            "  n2->sort = \"SortK{}\";\n" +
+            "  n->children[1] = n2;\n" +
+            "  value_type result = {.nterm = n};\n" +
+            "  $$ = result;\n" +
```

Comment:
Are these changes (to KSyntax2Bison.java) related to the goals of this PR (making Bison default?) If not, can we plul it out into its own PR?